src/ff/field.rs

```diff
@@ -119,6 +127,20 @@ pub trait Field:
     }
 }
 
+/// Mapping between a [`Field`]'s `TYPE_STR` and its `SIZE_IN_BYTES`
+/// # Errors
+/// If unknown `type_str` is passed in
+pub fn size_in_bytes_from_type_str(type_str: &str) -> Result<u32, Error> {
```

Comment:
If you define a new trait, then you can implement it for `&str` and then you avoid the oddly-named function.  Instead you get:

```rust
use ...::FieldTypeString;
F::TYPE_STR.size_in_bytes()
```